controllers/user.js

```diff
@@ -433,7 +433,9 @@ async function postRemoveFxm(req, res) {
     await DB.removeKan(sessionUser);
   }
   //END DATA REMOVAL SPECIFIC
-
+  if (req.session?.kanary) {
+    req.session.kanary.onRemovalPilotList = false;
```

Comment:
Q: Is it possible that _`kanary`_ could be `undefined`? Or is every `req.session` object guaranteed to have a `.kanary` property (and therefore trying to access `.onRemovalPilotList` property wouldn't error out).

Reply:
@pdehaan  - just learned this from Amri, but that `req.session?.kanary` is an [optional chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) - meaning if anything beyond the `req.session` object is undefined, that code doesn't execute. The code is an object, so setting `onRemovalPilotList` is fine (though maybe it'd be better to do `if req.session?.onRemovalPilotList`)

Reply:
Oh interesting, I'll have to poke at that. I always thought you need to do something like `req.session?.kanary?`.
But yeah, makes sense. Rereading the code it will see if `req.session?.kanary` is defined/truthy, so if it was `undefined` the `if() {...}` block would fail and not try and set the `.onRemovalPilotList` property anyways.

Reply:
<img width="496" alt="Screen Shot 2022-01-06 at 4 25 19 PM" src="https://user-images.githubusercontent.com/520541/148471294-8a8430df-a43b-4585-b092-80d5927a2478.png">